src/dataset.js

```diff
@@ -32,10 +32,15 @@ class State {
     return next;
   }
 
-  get records() {
-    return this.pages.reduce(function(records, page) {
-      return records.concat(page.records);
-    }, []);
+  get length() {
+    return this.totalSize;
```

Comment:
maybe make length a simple property? in otherwords `this.length` instead of `this.totalSize`